File: resource/interface/client/textui.lua
---@class TextUIOptions
---@field position? 'right-center' | 'left-center' | 'top-center' | 'bottom-center';
---@field icon? string | {[1]: IconProp, [2]: string};
---@field iconColor? string;
---@field style? string | table;
---@field alignIcon? 'top' | 'center';

local isOpen = false
local currentText

---@param text string
---@param options? TextUIOptions
function lib.showTextUI(text, options)
    if currentText == text then return end

    if not options then options = {} end

    options.text = text
    currentText = text

    SendNUIMessage({
        action = 'textUi',
        data = options
    })

    isOpen = true
end

function lib.hideTextUI()
    SendNUIMessage({
        action = 'textUiHide'
    })

    isOpen = false
    currentText = nil
end

---@return boolean, string | nil
function lib.isTextUIOpen()
    return isOpen, currentText
end

File: web/src/typings/textui.ts
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import React from 'react';

export type TextUiPosition = 'right-center' | 'left-center' | 'top-center';

export interface TextUiProps {
  text: string;
  position?: TextUiPosition;
  icon?: IconProp;
  iconColor?: string;
  style?: React.CSSProperties;
}
